src/main/scala/mesosphere/marathon/api/serialization/ContainerSerializer.scala

```diff
@@ -22,57 +22,18 @@ object ContainerSerializer {
     val maybeDocker = if (proto.hasDocker) Some(DockerSerializer.fromProto(proto.getDocker)) else None
     Container(
       `type` = proto.getType,
-      volumes = proto.getVolumesList.asScala.map(Volume(_)).to[Seq],
+      volumes = proto.getVolumesList.asScala.map(Volume.fromProto(_)).to[Seq],
       docker = maybeDocker
     )
   }
 
   def toMesos(container: Container): mesos.Protos.ContainerInfo = {
-    // we can only serialize DockerVolumes into a Mesos Protobuf.
-    // PersistentVolumes and ExternalVolumes are handled differently
-    val serializedVolumes = container.volumes.collect { case dv: DockerVolume => VolumeSerializer.toMesos(dv) }
-    val builder = mesos.Protos.ContainerInfo.newBuilder
-      .setType(container.`type`)
-      .addAllVolumes(serializedVolumes.asJava)
+    val builder = mesos.Protos.ContainerInfo.newBuilder.setType(container.`type`)
     container.docker.foreach { d => builder.setDocker(DockerSerializer.toMesos(d)) }
     builder.build
   }
 }
 
-object VolumeSerializer {
```

Comment:
You cannot know that but we want to separate the serialization logic from the model object. Serialization has nothing to do with the rest of the "business logic". Therefore, deleting this separate object is a step backwards.

I know that our code is currently inconsistent regarding this. The reason is that our `MarathonStore` currently still requires objects convertible to `MarathonState` objects that provide the serialization logic. It would be much better if it required a `MarathonConverter` instead which provided the serialization logic.